tests/integration/components/moderators/component-test.ts

```diff
@@ -0,0 +1,140 @@
+        assert.dom(`[data-test-moderator-row="${currentUser.id}"]>div>[data-test-permission-group]`).hasText('Admin');
+        assert.dom(`[data-test-moderator-row="${moderator.id}"]>div>[data-test-permission-group]`).hasText('Moderator');
+        await clickTrigger(`[data-test-moderator-row="${moderator.id}"]`);
+        await selectChoose(`[data-test-moderator-row="${moderator.id}"]`, 'Admin');
+        assert.dom(`[data-test-moderator-row="${currentUser.id}"]>div>[data-test-permission-group]`).hasText('Admin');
+        assert.dom(`[data-test-moderator-row="${moderator.id}"]>div>[data-test-permission-group]`).hasText('Admin');
+        await click(`[data-test-delete-moderator-button="${moderator.id}"]>[data-test-delete-button]`);
+        await click('[data-test-confirm-delete]');
+        assert.dom('[data-test-moderator-link]').exists({ count: 1 });
+        assert.dom('[data-test-permission-group]').exists({ count: 1 });
+        assert.dom('[data-test-delete-moderator-button]').exists({ count: 1 });
+    });
+
+    test('can only remove self as a moderator', async function(
+        this: ModeratorsTestContext, assert,
+    ) {
+        const currentUser = server.create('user', { id: 'sprout' });
+        const currentUserModel = await this.store.findRecord('user', 'sprout');
+        this.owner.lookup('service:current-user').setProperties({
+            user: currentUserModel, currentUserId: currentUserModel.id,
+        });
+        const admin = server.create('moderator', {
+            permissionGroup: PermissionGroup.Admin,
+        });
+        const moderator = server.create('moderator', {
+            id: currentUserModel.id,
+            user: currentUser,
+            permissionGroup: PermissionGroup.Moderator,
+        });
+        this.provider.update({
+            moderators: [moderator, admin],
+        });
+        await render(
+            hbs`<Moderators::Manager
+                    @provider={{this.providerModel}}
+                    as |moderatorManager|
+                >
+                    <Moderators::List
+                        @manager={{moderatorManager}}
+                    />
+                </Moderators::Manager>`,
+        );
+        assert.dom('[data-test-moderator-link]').exists({ count: 2 });
+        assert.dom('[data-test-permission-group]').exists({ count: 2 });
+        assert.dom('[data-test-delete-moderator-button]').exists({ count: 1 });
+        assert.dom(`[data-test-delete-moderator-button="${admin.id}"]`).doesNotExist();
+        assert.dom(
+            `[data-test-moderator-row="${currentUser.id}"]>div>[data-test-permission-group]`,
+        ).hasText('Moderator');
+        assert.dom(`[data-test-moderator-row="${admin.id}"]>div>[data-test-permission-group]`).hasText('Admin');
+        await click(`[data-test-delete-moderator-button="${moderator.id}"]>[data-test-delete-button]`);
+        await click('[data-test-confirm-delete]');
```

Comment:
Likewise here, I would check if the wording is clear in the modal that you are removing yourself as a moderator